force-app/main/default/lwc/rd2EntryForm/rd2EntryForm.js

```diff
@@ -441,124 +412,96 @@ export default class rd2EntryForm extends LightningElement {
+                payload: event.detail
+            });
     }
 
-    /***
-     * @description Get the Expected Value based on the current Amount, Paid Amount and Installment Number
-     */
-    getExpectedTotalValue(duringInit) {
-        const allFields = this.getAllFields();
-        const amount = duringInit ? getFieldValue(this.record, FIELD_AMOUNT)
-            : allFields[FIELD_AMOUNT.fieldApiName];
-        const paidAmount = this.returnZeroIfNull(getFieldValue(this.record, FIELD_PAID_AMOUNT));
-        const paidInstallments = this.returnZeroIfNull(getFieldValue(this.record, FIELD_PAID_INSTALLMENTS));
-        const numberOfInstallments = duringInit ? getFieldValue(this.record, FIELD_INSTALLMENTS)
-            : allFields[FIELD_INSTALLMENTS.fieldApiName];
-        const remainingInstallments = numberOfInstallments - paidInstallments;
-        const expectedValue = paidAmount + remainingInstallments * amount;
-        return expectedValue;
-    }
 
     /***
-     * @description Returns 0 if the provided value is null
+     * @description Currency change might hide or display the credit card widget
+     * @param event
      */
-    returnZeroIfNull(numberField) {
-        return numberField != null ? numberField : 0;
+    handleCurrencyChange(event) {
+        this.rd2State = this.rd2Service.dispatch(this.rd2State, {
+            type: ACTIONS.SET_CURRENCY,
+            payload: event.target.value
+        });
+        this.handleElevateWidgetDisplay();
     }
 
     /***
-     * @description Returns the Annual Value for Open Donations and Expected Value for Fixed
+     * @description Checks if the credit card widget should be displayed.
      */
-    returnDonationValue(duringInit) {
-        if (this.getRecurringType() === RECURRING_TYPE_OPEN) {
-            return this.getAnnualValue(duringInit);
-        } else if (this.getRecurringType() === RECURRING_TYPE_FIXED) {
-            return this.getExpectedTotalValue(duringInit);
+    handleElevateWidgetDisplay() {
+        if (this.isElevateCustomer) {
+            this.evaluateElevateWidget();
         }
     }
 
-    /***
-     * @description Checks if the Annual or Expected Total Value changed, which will update the Change Type
-     */
-    checkForDonationValueChange() {
-        let newChangeType = "";
-        let newdonationValue = this.returnDonationValue();
-
-        if (this.recurringType !== this.getRecurringType()) {
-            newChangeType = "";
-        } else if (newdonationValue > this.donationValue) {
-            newChangeType = CHANGE_TYPE_UPGRADE;
-        } else if (newdonationValue < this.donationValue) {
-            newChangeType = CHANGE_TYPE_DOWNGRADE;
-        }
+    handleAmountChange(event) {
+        this.rd2State = this.rd2Service.dispatch(this.rd2State, {
+            type: ACTIONS.SET_DONATION_AMOUNT,
+            payload: event.target.value
+        });
+    }
 
-        this.changeType = newChangeType;
+    get _showChangeTypeField() {
```

Comment:
I see you have a TODO in the LWC style guide to talk about this naming convention, I think that's worth doing since it isn't common in NPSP.

Reply:
I'm going to pull these underscores from these getters, that naming convention is for variables that are named the same as their getters. We probably shouldn't have leading underscores as part of the name for a getter.